The purpose of this assignment was to upload to GitHub a Claude integrated jupyternotebook environment and create a query function that takes questions as string inputs,feeds it to Claude and outputs a response within the jupyternotebook. The Claude model integrated into the Jupyter notebook for all queries was the claude-haiku-4-5-20251001".
Three physics problems from quantum mechanics and astrophysics textbooks[1][2]were placed as inputs in the query function[3][4], and the outputted numerical result has been featured in the document and compared with the true calculated results from their respective textbooks. The discrepancy percentage was calculated using this formula below. 
percent_error = (abs(C - T) / T)* 100
C = Claude Value B = Textbook Value
The contents of this document are the results of rerunning the jupyternotebook script four separate times to ensure the accuracy/consistency of the statements made in this document.


In [6]:
import os
from dotenv import load_dotenv
from anthropic import Anthropic
load_dotenv()

True

In [5]:
client = Anthropic()  # automatically reads ANTHROPIC_API_KEY from environment

In [3]:
def query_llm(prompt: str, model: str = 'claude-haiku-4-5-20251001'):
    """ This function takes a prompt input typed in by the user as a string and feeds the prompt to the Claude Haiku 4.5 LLM and returns a text response as a result"""
    client = Anthropic(api_key=os.environ.get("ANTHROPIC_API_KEY"))
    if not os.environ.get("ANTHROPIC_API_KEY"):
        print("Set ANTHROPIC_API_KEY to run this cell.")
    else:
        message = client.messages.create(
            model= model,
            max_tokens=2048,
            messages=[
                {"role": "user",
                 "content": prompt,
                }
            ],
        )
    for block in message.content:
        if block.type == "text":
            print(block.text)

In [4]:
question_1 = """When two ultraviolet beams of wavelength ( lambda_1 = 80nm and lambda_2 = 110nm) 
fall on a lead surface they produce photoelectrons with maximum energies of 11.390eV and 7.154eV respectfully. 
Estimate the numerical value of Planck's constant given these values.Calculate the cutoff frequency and wavelength of lead."""
#Page 22 of Quantum Mechanics  Concepts and Applications 3rd Edition Nourdine zettili Wiley 
response_physics_question_1 = query_llm(question_1)

# Solving for Planck's Constant and Lead's Photoelectric Properties

## Given Information
- λ₁ = 80 nm, KE₁(max) = 11.390 eV
- λ₂ = 110 nm, KE₂(max) = 7.154 eV

## Part 1: Finding Planck's Constant

Using the photoelectric equation:
$$KE_{max} = h\nu - \phi = \frac{hc}{\lambda} - \phi$$

For the two beams:
$$\frac{hc}{\lambda_1} - \phi = KE_1 \quad \text{...(1)}$$
$$\frac{hc}{\lambda_2} - \phi = KE_2 \quad \text{...(2)}$$

Subtracting equation (2) from equation (1):
$$hc\left(\frac{1}{\lambda_1} - \frac{1}{\lambda_2}\right) = KE_1 - KE_2$$

$$h = \frac{KE_1 - KE_2}{c\left(\frac{1}{\lambda_1} - \frac{1}{\lambda_2}\right)}$$

**Calculating the numerator:**
$$KE_1 - KE_2 = 11.390 - 7.154 = 4.236 \text{ eV}$$

**Calculating the denominator:**
$$\frac{1}{\lambda_1} - \frac{1}{\lambda_2} = \frac{1}{80 \times 10^{-9}} - \frac{1}{110 \times 10^{-9}}$$
$$= 1.25 \times 10^7 - 9.091 \times 10^6 = 3.409 \times 10^6 \text{ m}^{-1}$$

**Computing h:**
$$h = \frac{4.236 \text{ eV}}{3 \times 10^8 \tex

### ACCEPTED TEXTBOOK VALUE AND DISCREPENCY
ACCEPTED TEXTBOOK VALUE AND DISCREPENCY: 
The Quantum Mechanics textbook used for this assignment [1] states that the calculated parameters given the provided values are as follows: 
| Parameter |Textbook Value | Percent Discrepency 

|-----------|-------| 
| Planck's Constant (h) | 6.63 × 10⁻³⁴ J·s | Claude Result Discrepancy: 0.0% 


| Work Function (φ) | 4.12 eV | Claude Result Discrepancy: 0.48% 


| Cutoff Frequency (ν₀) |10^15 Hz | Claude Result Discrepancy: 0.7% 


| Cutoff Wavelength (λ₀) | 300 nm (UV region) | Claude Result Discrepancy: 0.6% 


The percent discrepancy values for Planck’s constant, Work Function Cutoff Frequency and Cutoff Wavelength are small enough for the Claude output to be considered correct.



In [5]:
question_2 = """Assuming for simplicity that the Moon's orbit is exactly circular, 
we can calculate the centripetal acceleration of the Moon rapidly.
In the case r is the distance from the center of Earth to the center of the moon,
r = 3.84401x10^8m, and v is the Moon's orbital velocity is given by v. P is the period 27.3 days
find the acceleration of the moon."""
# page 36 Introduction to Modern Astrophysics 2nd Edition Bradley W. Carroll and Dale A. Ostlie 
response_physics_question_2 = query_llm(question_2)

# Calculating the Moon's Centripetal Acceleration

## Given Information
- Orbital radius: r = 3.84401 × 10⁸ m
- Period: P = 27.3 days
- Orbit: circular

## Solution

### Step 1: Find the Moon's Orbital Velocity

The Moon travels the circumference of its orbit in one period:

$$v = \frac{2\pi r}{P}$$

First, convert the period to seconds:
$$P = 27.3 \text{ days} × 86,400 \text{ s/day} = 2.358 × 10^6 \text{ s}$$

Now calculate velocity:
$$v = \frac{2\pi (3.84401 × 10^8)}{2.358 × 10^6}$$

$$v = \frac{2.415 × 10^9}{2.358 × 10^6} = 1,024 \text{ m/s}$$

### Step 2: Calculate Centripetal Acceleration

For circular motion:
$$a_c = \frac{v^2}{r}$$

$$a_c = \frac{(1,024)^2}{3.84401 × 10^8}$$

$$a_c = \frac{1.049 × 10^6}{3.84401 × 10^8}$$

$$\boxed{a_c = 0.00273 \text{ m/s}^2 = 2.73 × 10^{-3} \text{ m/s}^2}$$

Or equivalently, using the direct formula:

$$a_c = \frac{4\pi^2 r}{P^2} = \frac{4\pi^2(3.84401 × 10^8)}{(2.358 × 10^6)^2} = 2.73 × 10^{-3} \text{ m/s}^2$$

**The Moon's centripetal acceler

### ACCEPTED TEXTBOOK VALUE AND DISCREPENCY
The Introduction to Astrophysics textbook value used for the moon's centripetal acceleration [2] calculated based on the given parameters is 0.0027 ms-2 resulting in a 1.1% percent discrepancy between the true textbook value and the Claude response value of 0.00273 ms-2; therefore, this Claude response is considered correct and the discrepancy can be attributed to how the textbook decided to round the values.


In [6]:
question_3a = "what is the maximum angular momentum of a solar-mass black hole in SI units?"
# page 640 stars textbook example 17.3.2 Introduction to Modern Astrophysics 2nd Edition Bradley W. Carroll and Dale A. Ostlie 
response_physics_question_3a = query_llm(question_3a)

# Maximum Angular Momentum of a Solar-Mass Black Hole

For a maximally rotating (Kerr) black hole, the angular momentum is:

$$L_{\max} = \frac{GM^2}{c}$$

## Calculation

Using:
- $M = M_{\odot} = 1.989 \times 10^{30}$ kg
- $G = 6.674 \times 10^{-11}$ m³ kg⁻¹ s⁻²
- $c = 2.998 \times 10^8$ m/s

$$L_{\max} = \frac{(6.674 \times 10^{-11})(1.989 \times 10^{30})^2}{2.998 \times 10^8}$$

$$L_{\max} \approx 7.0 \times 10^{50} \text{ J·s}$$

Or equivalently: **~7 × 10⁵⁰ kg·m²/s**

This corresponds to a dimensionless spin parameter $a^* = 1$, where the ergosphere reaches its maximum extent and the black hole rotates at the speed of light at its event horizon.


### ACCEPTED TEXTBOOK VALUE AND DISCREPENCY
The true value given by the introduction to astrophysics textbook and by inputting the parameters into the formula is 8.81E41 kgm^2s^-1. The formula provided by Claude as well as the values it supplied for the mass of the sun, gravitational constant and speed of light are all correct; however, this Claude calculated value of 7E50 kgm^2s^-1 whichis clearly a hallucination/incorrect as the percentage of discrepancy is 7.95E10%. This is a direct result of how Claude uses an attention method to calculate/ output its next result, as each key/query was being produced/generated there must have been an issue with how the exponents were multiplied together resulting in orders of magnitude discrepencies between the true value and the claude ouput. The probability function towards the end of this key/query matrix calculations also must have been stretched very thin as the output varies between 1E48 and 1E50 much larger than the true values order of magnitude 1e41.


### REFERENCES 

[1] Zettili, N. (2009). Quantum mechanics: Concepts and Applications (3rd ed., p. 22). Wiley. 

[2] Carroll, B. W., & Ostlie, D. A. (2007). Introduction to modern astrophysics (2nd ed., p. 36). Pearson. 

[3] OpenAI. (2026). Developer quickstart. Retrieved from https://platform.openai.com/docs/quickstart

[4] Anthropic. (2026). Claude (Version Haiku 4.5) [Large language model]. https://claude.ai

